### PR TITLE
Binder

### PR DESCRIPTION
**Pull request by sjsrey**
Building a binder for the tutorial.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -8,7 +8,12 @@ dependencies:
...
+  - libpysal
+  - matplotlib
   - jupyter
   - jupyterlab
   - nbconvert
-  - brewer2mpl
+  - palettable
+  - descartes
+  - pip:
+    - git+https://github.com/sjsrey/mapclassify.git@binder
```
Comment (jGaboardi) [inline environment.yml:19]: * Should this be `pysal/mapclassify.git@binder`?
* Where is the binder being launched from? Shouldn't there be a badge or something in the `README.md`?

Comment (sjsrey) [reply environment.yml:19]: Yeah. I needed to get it working first on my fork. 
Once this is merged we can get the badge for upstream.

Comment (jGaboardi) [reply environment.yml:19]: * binder launches successfully 
* notebooks run successfully:
  * `01_maximum_breaks.ipynb`
  * `02_legends.ipynb`
  * `03_choropleth.ipynb`
  * `04_pooled.ipynb`
* notebooks with errors:
  * `05_Greedy_coloring.ipynb`

The following traceback is produced from cell 3 in `05_Greedy_coloring.ipynb`; looks like we need to add `networkx` into `environment.yml` (and also `requirements.txt` and `mapclassify.yml`?):

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/srv/conda/envs/notebook/lib/python3.6/site-packages/mapclassify/greedy.py in greedy(gdf, strategy, balance, min_colors, sw, min_distance, silence_warnings, interchange)
    287     try:
--> 288         import networkx as nx
    289     except ImportError:

ModuleNotFoundError: No module named 'networkx'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-3-6603cfcd9243> in <module>
----> 1 africa['greedy_default'] = greedy(africa)
      2 africa['greedy_default'].head(5)

/srv/conda/envs/notebook/lib/python3.6/site-packages/mapclassify/greedy.py in greedy(gdf, strategy, balance, min_colors, sw, min_distance, silence_warnings, interchange)
    288         import networkx as nx
    289     except ImportError:
--> 290         raise ImportError("The 'networkx' package is required.")
    291     try:
    292         import pandas as pd

ImportError: The 'networkx' package is required.
```

Comment (jGaboardi) [reply environment.yml:20]: Need to add `seaborn`, too (cell 9 in `05_Greedy_coloring.ipynb`):

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-9-2edc5b0bfc23> in <module>
      2 import numpy as np
      3 from time import time
----> 4 import seaborn as sns
      5 import matplotlib.pyplot as plt
      6 

ModuleNotFoundError: No module named 'seaborn'
```

**Comment by jGaboardi** [comment]
[`networkx` is required for `greedy`](https://github.com/pysal/mapclassify/blob/0b9eb7d784042a24e8eb5848b5287995a753e95a/mapclassify/greedy.py#L287-L290) and should be added to `requirements.txt`.

```python
    try:
        import networkx as nx
    except ImportError:
        raise ImportError("The 'networkx' package is required.")
```

**Comment by jGaboardi** [comment]
Where is [`mapclassify.yml `](https://github.com/pysal/mapclassify/blob/master/mapclassify.yml) being used?

File: environment.yml (hunk 1 above)
Comment (jGaboardi) [reply environment.yml:21]: `05_Greedy_coloring.ipynb` now runs successfully.

**Pull request merged**